test/cosineSimilarityEngineCondition.test.ts

```diff
@@ -0,0 +1,291 @@
+import { check } from '../src/engine';
+import {
+  EngineCondition,
+  QueryFilterComparisonType,
+  CosineSimilarityFilter,
+  PageView
+} from '../types';
+
+/* TODO queryProperty based tests for engine behaviors
```

Comment:
Could you please open a new issue to discuss this TODO?

Reply:
This should have been `property based tests for engine behaviors`.

I'm going to write a minimal example to discuss this topic further.